pkg/admission/webhook/generic_webhook.go

```diff
@@ -160,10 +160,7 @@ func (p *WebhookDispatcher) getAPIExportCluster(attr admission.Attributes, clust
 	for _, apiBinding := range objs {
 		for _, br := range apiBinding.Status.BoundResources {
 			if br.Group == attr.GetResource().Group && br.Resource == attr.GetResource().Resource {
-				path := logicalcluster.NewPath(apiBinding.Spec.Reference.Export.Path)
-				if path.Empty() {
-					path = clusterName.Path()
-				}
+				path := logicalcluster.NewPath(apiBinding.Status.LogicalCluster)
```

Comment:
This needs to be a `logicalcluster.Name` and you'll need to change the implementation of `getAPIExport` to get directly from the lister instead of using the indexer+index.